### PR TITLE
Feature/farm 50 update farm information

### PR DESCRIPTION
**Pull request by LeeMatthewHiggins**
[FARM-50 - ]

#### 📲 What

Added transformer to hook up the chat answers to the recommendation engine

#### 🤔 Why
		
this is needed to give good recommendation percentages and to be able to test recommendation engine.

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: lib/model/bloc/startup/ChatResponseToPlotInfoTransformer.dart [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  final dateFormat = DateFormat(_Strings.dateFormat);
+
+  @override
+  Map<String,String> transform({Map<String,ChatResponseViewModel> from}) {
+    return from.map((key,value){
+      final date = castOrNull<DateTime>(value.value);
+      if (date != null){
+        return MapEntry(key, dateFormat.format(date));
+      }
+      return MapEntry(key, value.id);
```
Comment (albertmiro) [inline lib/model/bloc/startup/ChatResponseToPlotInfoTransformer.dart]: Another way to do that:
```
val mapValue = (date != null) ? dateFormat.format(date) : value.id;
return MapEntry(key, mapValue);
```

Take the one you like most ;-)

File: lib/model/bloc/startup/StartupViewModelProvider.dart [hunk 2]
```diff
@@ -65,21 +68,34 @@ class StartupViewModelProvider implements ViewModelProvider<StartupViewModel> {
...
+            plotInfo,
+          );
+          account.profileRepository.add(newProfile).then((profile) {
+            account.profileRepository.switchTo(profile);
+            _refresh();
+          });
         });
+      }
+    }, (data) {
+      //TODO: error case, should display a popup.
```
Comment (albertmiro) [inline lib/model/bloc/startup/StartupViewModelProvider.dart:79]: This would be done in another PR?

File: lib/ui/LandingPage.dart [hunk 3]
```diff
@@ -134,7 +134,8 @@ class LandingPage extends StatelessWidget {
 
   @override
```
Comment (albertmiro) [inline lib/ui/LandingPage.dart]: That has been done also in https://github.com/farmsmart/farmsmart-flutter/pull/109, isn't it? I think that this class should be removed from this PR.

**Review by albertmiro — CHANGES_REQUESTED** [review]

File: lib/model/bloc/startup/ChatResponseToPlotInfoTransformer.dart [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  static const dateFormat = "MMMM";
+}
+
+class ChatResponseToPlotInfoTransformer extends ObjectTransformer<Map<String,ChatResponseViewModel>, Map<String,String>> {
+
+  final dateFormat = DateFormat(_Strings.dateFormat);
+
+  @override
+  Map<String,String> transform({Map<String,ChatResponseViewModel> from}) {
+    return from.map((key,value){
```
Comment (alexanishchenko88) [inline lib/model/bloc/startup/ChatResponseToPlotInfoTransformer.dart]: Maybe change this alias naming to something like **responseViewModel** or **response** or **viewModel** just to avoid this kind of invocation 
**value.value**

File: lib/model/bloc/startup/StartupViewModelProvider.dart [hunk 5]
```diff
@@ -65,21 +68,34 @@ class StartupViewModelProvider implements ViewModelProvider<StartupViewModel> {
...
-                account.profileRepository.switchTo(profile);
-                _refresh();
-              }) ;
-            });
-          }
-        },(data) {
-            //TODO: error case, should display a popup.
+    //TODO: Remove the Mock ID´s once implemented
+    final transformer = ChatResponseToPlotInfoTransformer();
+    return ChatPageViewModel(_LocalisedAssets.onboardingFlow(), (data) {
```
Comment (alexanishchenko88) [inline lib/model/bloc/startup/StartupViewModelProvider.dart:72]: This return does a lot of stuff inside, I think it should be split into some private methods

File: lib/model/bloc/startup/ChatResponseToPlotInfoTransformer.dart (hunk 1 above)
Comment (LeeMatthewHiggins) [reply lib/model/bloc/startup/ChatResponseToPlotInfoTransformer.dart]: this will expand with more types I expect, so I leave longhand for now

File: lib/ui/LandingPage.dart (hunk 3 above)
Comment (LeeMatthewHiggins) [reply lib/ui/LandingPage.dart]: rebased, this should be removed from the PR now.

File: lib/model/bloc/startup/StartupViewModelProvider.dart (hunk 2 above)
Comment (LeeMatthewHiggins) [reply lib/model/bloc/startup/StartupViewModelProvider.dart:79]: yes, not sure how to deal with this ATM. not sure what an error would be. If the user quit, then they would just land back at the screen they started, so we don't need to do anything.

File: lib/model/bloc/startup/ChatResponseToPlotInfoTransformer.dart (hunk 4 above)
Comment (LeeMatthewHiggins) [reply lib/model/bloc/startup/ChatResponseToPlotInfoTransformer.dart]: 👍

**Review by alexanishchenko88 — APPROVED** [review]

File: lib/model/bloc/startup/StartupViewModelProvider.dart (hunk 5 above)
Comment (LeeMatthewHiggins) [reply lib/model/bloc/startup/StartupViewModelProvider.dart:72]: split this up a bit, it will change in another PR too

**Review by albertmiro — APPROVED** [review]
